tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -148,38 +160,10 @@ public void release(T element) {
    * @param element
    */
   public void discard(T element) {
```

Comment:
Probably because there's no way to detect if a thrift client is in a bad state, just based upon inspection of the object.
So only the client side code can know if an exception happened and its potentially in a bad state.